app/models/generic_object.rb

```diff
@@ -78,6 +69,6 @@ def ca_attribute_for_inspect(value)
   end
 
   def type_cast(attr_name, value)
-    TYPE_MAP.fetch(generic_object_definition.defined_attributes[attr_name].to_sym).cast(value)
+    GenericObjectDefinition::TYPE_MAP.fetch(generic_object_definition.defined_attributes[attr_name]).cast(value)
```

Comment:
If the TYPE_MAP lives on the GenericObjectDefintion, perhaps the `type_cast` method should live there as well since they are intrinsically tied together.

So if type_cast where there, I would expect something like...

```ruby
# On GenericObject
delegate :type_cast, :to => :generic_object_defintion

# On GenericObjectDefinition
def type_cast(attr_name, value)
  TYPE_MAP.fetch(defined_attributes[attr_name]).cast(value)
end
```

You may not even need the delegation...depends on how type_cast is called.